code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentManagerImpl.java

```diff
@@ -295,7 +295,7 @@ public void serviceUpdate(ConfigUpdate update) {
             @Override
             public void run() {
                 Service service = objectMgr.loadResource(Service.class, client.getResourceId());
-                if (service != null && service.getState().equalsIgnoreCase(CommonStatesConstants.ACTIVE)) {
+                if (service != null && sdSvc.isActiveService(service)) {
```

Comment:
this method returns true if service is in any of activIsh states (updating-active, upgrading, restarting, etc)